Keyring is accessed even if password is passed on the command line

ref https://github.com/kopia/kopia/issues/732

```
> "kopia.exe" snapshot create --password=[MASKED] --description "asdfg" "C:\some\path" (In: false, Out: false, Err: false)
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x38 pc=0x140cc7a]
goroutine 1 [running]:
github.com/zalando/go-keyring.windowsKeychain.Get(0xc000698000, 0x22, 0xc00015f8ee, 0x8, 0x1aeeba0, 0xc00015f8e0, 0x1adb740, 0xc000694020)
	/home/travis/gopath/pkg/mod/github.com/zalando/go-keyring@v0.1.0/keyring_windows.go:20 +0x11a
github.com/zalando/go-keyring.Get(...)
	/home/travis/gopath/pkg/mod/github.com/zalando/go-keyring@v0.1.0/keyring.go:32
github.com/kopia/kopia/repo.GetPersistedPassword(0x1e2fbc0, 0xc0004f0f00, 0xc0004606c0, 0x39, 0xc0004eb460, 0x0, 0xc0004f0f30)
	/home/travis/gopath/src/github.com/kopia/kopia/repo/password.go:26 +0x375
github.com/kopia/kopia/cli.getPasswordFromFlags(0x1e2fbc0, 0xc0004f0f00, 0x1e10100, 0xc0000660d0, 0x0, 0x0, 0x1aecd60)
	/home/travis/gopath/src/github.com/kopia/kopia/cli/password.go:56 +0x145
github.com/kopia/kopia/cli.openRepository(0x1e2fbc0, 0xc0004f0f00, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kopia/kopia/cli/config.go:60 +0x165
github.com/kopia/kopia/cli.maybeRepositoryAction.func1.1(0x0, 0x0)
	/home/travis/gopath/src/github.com/kopia/kopia/cli/app.go:141 +0xf0
github.com/kopia/kopia/cli.withProfiling(...)
	/home/travis/gopath/src/github.com/kopia/kopia/cli/profile_disabled.go:7
github.com/kopia/kopia/cli.maybeRepositoryAction.func1(0xc000160d80, 0xf2bcaa, 0x1b645e0)
	/home/travis/gopath/src/github.com/kopia/kopia/cli/app.go:125 +0x64
gopkg.in/alecthomas/kingpin%2ev2.(*actionMixin).applyActions(0xc0004d1458, 0xc000160d80, 0x0, 0x0)
	/home/travis/gopath/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/actions.go:28 +0x74
gopkg.in/alecthomas/kingpin%2ev2.(*Application).applyActions(0xc0001460f0, 0xc000160d80, 0x0, 0x0)
	/home/travis/gopath/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:557 +0xe3
gopkg.in/alecthomas/kingpin%2ev2.(*Application).execute(0xc0001460f0, 0xc000160d80, 0xc0004eb2a0, 0x2, 0x2, 0x0, 0x0, 0x0, 0xc000007d60)
	/home/travis/gopath/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:390 +0xa5
gopkg.in/alecthomas/kingpin%2ev2.(*Application).Parse(0xc0001460f0, 0xc000126010, 0x6, 0x7, 0x1, 0xc000007d68, 0x0, 0x1)
	/home/travis/gopath/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:222 +0x213
main.main()
	/home/travis/gopath/src/github.com/kopia/kopia/main.go:77 +0x188
```

I would expect that if I pass the repository password on the command line, it would not need to go to the credential store.

--- a/cli/password.go
+++ b/cli/password.go
@@ -47,26 +47,26 @@ func askForExistingRepositoryPassword() (string, error) {
 var passwordFromToken string
 
 func getPasswordFromFlags(ctx context.Context, isNew, allowPersistent bool) (string, error) {
-	if passwordFromToken != "" {
-		// password provided via token
+	switch {
+	case passwordFromToken != "":
+		// password extracted from connection token
 		return passwordFromToken, nil
-	}
-
-	if !isNew && allowPersistent {
+	case *password != "":
+		// password provided via --password flag or KOPIA_PASSWORD environment variable
+		return strings.TrimSpace(*password), nil
+	case isNew:
+		// this is a new repository, ask for password
+		return askForNewRepositoryPassword()
+	case allowPersistent:
+		// try fetching the password from persistent storage specific to the configuration file.
 		pass, ok := repo.GetPersistedPassword(ctx, repositoryConfigFileName())
 		if ok {
 			return pass, nil
 		}
 	}
 
-	switch {
-	case *password != "":
-		return strings.TrimSpace(*password), nil
-	case isNew:
-		return askForNewRepositoryPassword()
-	default:
-		return askForExistingRepositoryPassword()
-	}
+	// fall back to asking for existing password
+	return askForExistingRepositoryPassword()
 }
 
 // askPass presents a given prompt and asks the user for password.

--- a/tests/end_to_end_test/api_server_repository_test.go
+++ b/tests/end_to_end_test/api_server_repository_test.go
@@ -75,6 +75,11 @@ func TestAPIServerRepository(t *testing.T) {
 		"--password", "baz",
 	)
 
+	// we are providing custom password to connect, make sure we won't be providing
+	// (different) default password via environment variable, as command-line password
+	// takes precedence over persisted password.
+	e2.RemoveDefaultPassword()
+
 	// should see one snapshot
 	snapshots := e2.ListSnapshotsAndExpectSuccess(t)
 	if got, want := len(snapshots), 1; got != want {

--- a/tests/testenv/cli_test_env.go
+++ b/tests/testenv/cli_test_env.go
@@ -129,6 +129,19 @@ func NewCLITest(t *testing.T) *CLITest {
 	}
 }
 
+// RemoveDefaultPassword prevents KOPIA_PASSWORD from being passed to kopia.
+func (e *CLITest) RemoveDefaultPassword() {
+	var newEnv []string
+
+	for _, s := range e.Environment {
+		if !strings.HasPrefix(s, "KOPIA_PASSWORD=") {
+			newEnv = append(newEnv, s)
+		}
+	}
+
+	e.Environment = newEnv
+}
+
 // RunAndExpectSuccess runs the given command, expects it to succeed and returns its output lines.
 func (e *CLITest) RunAndExpectSuccess(t *testing.T, args ...string) []string {
 	t.Helper()

